cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -163,6 +170,12 @@ protected void startUp() throws Exception {
     }
 
     cancelHttpService = startHttpService(httpServiceBuilder.build());
+    TransactionRunners.run(transactionRunner, context -> {
+      AppMetadataStore appMetadataStore  = AppMetadataStore.create(context);
+      long count = appMetadataStore.getApplicationCount();
+      metricsCollectionService.getContext(Collections.emptyMap()).gauge(Constants.Metrics.Program.APPLICATION_COUNT,
```

Comment:
Emit the metrics outside the of the tx to minimize the tx scope. You can use the one that takes `TxCallable` too return the count from inside the transaction.

Reply:
fixed